### PR TITLE
docs: explain how to configure the dagger engine

### PR DESCRIPTION
**Pull request by TomChv**
Explain how to configure mirror on the dagger engine

Resolves: #5055

### CODE REVIEW

**Review by gerhard — COMMENTED** [review]
LGTM!

Want to approve & merge @vikram-dagger?

cc @shykes

File: docs/current/guides/967042-customize-engine-configuration.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+tags: ["docker"]
+authors: ["Tom Chauveau"]
+date: "2023-09-26"
+---
+
+# Customize Dagger Engine configuration
+
+## Introduction
+
+This guide explains how to configure Dagger Engine through the custom `engine.toml` file.
```
Comment (vikram-dagger) [inline docs/current/guides/967042-customize-engine-configuration.md:14]:
```suggestion
This guide explains how to configure the Dagger Engine through the custom `engine.toml` file.
```

File: docs/current/guides/967042-customize-engine-configuration.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+For example, this can be used to configure a registry mirror for pulling container images instead of DockerHub.
+
```
Comment (vikram-dagger) [inline docs/current/guides/967042-customize-engine-configuration.md:17]: Can you please add a "requirements" section here? similar to https://docs.dagger.io/882813/build-test-publish-java-spring#requirements, updating as needed.

File: docs/current/guides/967042-customize-engine-configuration.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+## Examples
+
+### Registry mirror
+
+Create a file `engine.toml` that contains the registry mirror, e.g.:
```
Comment (vikram-dagger) [inline docs/current/guides/967042-customize-engine-configuration.md:22]: I suggest renumbering this as a list of steps.

**Review by vikram-dagger — COMMENTED** [review]
See general feedback. Once you review and update, I'll do a copy edit on the text.

File: docs/current/guides/967042-customize-engine-configuration.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+```
+
+:::tip
+[See all Dagger Engine configuration options](https://docs.docker.com/build/buildkit/toml-configuration/).
+:::
+
+## Conclusion
+
+This guide described one way to customize the Dagger Engine configuration.
+Learn how to use [alternative runtimes for Dagger Engine](541047-alternative-runtimes.md), e.g. Podman, Kubernetes, etc.
```
[the remark this answers is not in the document]
Comment (TomChv) [reply docs/current/guides/967042-customize-engine-configuration.md:53]: I meant if you deploy the dagger engine in a Kubernetes cluster

File: docs/current/guides/967042-customize-engine-configuration.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+docker run --rm --name customized-dagger-engine --privileged --volume $PWD/engine.toml:/etc/dagger/engine.toml registry.dagger.io/engine:v0.8.7
+```
+
+Use this customised Dagger Engine:
+
+```shell
+export _EXPERIMENTAL_DAGGER_RUNNER_HOST=docker-container://customized-dagger-engine
+dagger query --progress=plain <<< '{ container { from(address:"hello-world") { stdout } } }'
+```
+
```
[the remark this answers is not in the document]
Comment (TomChv) [reply docs/current/guides/967042-customize-engine-configuration.md:45]: How? If you export this env variable, the customized engine can only be used

[the remark this answers is not in the document]
Comment (vikram-dagger) [reply docs/current/guides/967042-customize-engine-configuration.md]: Added

File: docs/current/guides/967042-customize-engine-configuration.md (hunk 2 above)
Comment (vikram-dagger) [reply docs/current/guides/967042-customize-engine-configuration.md]: Added

**Review by vikram-dagger — APPROVED** [review]

**Comment by vikram-dagger** [comment]
@TomChv, please see my changes and merge if OK.

File: docs/current/guides/967042-modify-engine-configuration.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  ```
+
+1. Test the configuration:
+
+  ```shell
+  export _EXPERIMENTAL_DAGGER_RUNNER_HOST=docker-container://customized-dagger-engine
+  dagger query --progress=plain <<< '{ container { from(address:"hello-world") { stdout } } }'
+  ```
+
+  You should see the specified `hello-world` container being pulled from the mirror instead of from Docker Hub.
```
Comment (vikram-dagger) [inline docs/current/guides/967042-modify-engine-configuration.md:48]: @TomChv I added this but please confirm this is what should happen or modify if wrong.

**Comment by sipsma** [comment]
I'm not totally sure if it's a good idea to put this in our "official website docs". The reason it's just in the "advanced" operator manual on our repo is that it's currently a pretty advanced use case that requires custom setup of Dagger, use of `_EXPERIMENTAL_*` env vars etc. We've generally tried to avoid going into all of that on our official docs since it's all meant to be a temporary solution to unblock power users, with better interfaces to come later.

For example, this guide is currently highly specific to a custom engine running in docker, but most of the time power users are running somewhere else, so it's not clear if the docker-specific instructions would be all that useful to them; I fear it may just raise more questions than it answers in addition to enshrining our current hacky OX as the "official" way.

@shykes curious to get your thoughts here

**Comment by TomChv** [comment]
@sipsma Should we put it into the operator manual then?

**Comment by TomChv** [comment]
@vikram-dagger @sipsma I updated this PR by removing the guide and instead added a small line in the FAQ that redirect to our operator guide.
I think it's still important to mention it on our doc so user can find a result if they use the search bar but same as rootless, most of the content is in the operator guide.

Let me know if there's anything to change.

@vikram-dagger This change lighten the complete exemple but we have no guided example in our operator manual, so I just added a small one line example to show how to replace the default config file with the user's one.

File: docs/current/faq.md [hunk 7]
```diff
@@ -170,6 +170,10 @@ Operations are automatically cached every time a Dagger pipeline runs. [Cache vo
 
 No. "Rootless mode" means running the Dagger Engine as a container without the `--privileged` flag. In this case, the container would not run as the "root" user of the system. Currently, the Dagger Engine cannot be run as a rootless container; network and filesystem constraints related to rootless usage would significantly limit its capabilities and performance. [Read more about these constraints](https://github.com/dagger/dagger/blob/main/core/docs/d7yxc-operator_manual.md).
 
+### Can I configure the Dagger Engine Runner?
```
[the remark this answers is not in the document]
Comment (TomChv) [reply docs/current/faq.md:173]: That's right, I'll update it today

**Review by sipsma — APPROVED** [review]
One nit I think is worth addressing, otherwise LGTM.

**Review by sipsma — APPROVED** [review]

**Pull request merged**